File: MAGMA/sgesv.c
#include <stdio.h>
#include <stdlib.h>
#include <magma.h>

int main()
{
  float *b, *AT;
  float A[3][3];
  int a = 0;
  magma_int_t* pivot;
  magma_int_t i = 0, j = 0, ok, status;
  magma_int_t m = 3, n = 1, mm = 9, mn = 3;
  magma_int_t info;

  AT = (float*)malloc(9*sizeof(int));
  b = (float*)malloc(3*sizeof(int));
  pivot = (magma_int_t*)malloc(3*sizeof(int));


  A[0][0] = 3.1;
  A[0][1] = 1.3;
  A[0][2] = -5.7;
  A[1][0] = 1;
  A[1][1] = -6.9;
  A[1][2] = 5.8;
  A[2][0] = 3.4;
  A[2][1] = 7.2;
  A[2][2] = -8.8;

  b[0] = -1.3;
  b[1] = -0.1;
  b[2] = 1.8;
  for(a = 0; a<3; a++)
    for(j = 0; j<3; j++)
      AT[j+3*a]=A[j][a];

  magma_sgesv(m, n, AT, m, pivot, b, m, &info);

  for (i = 0; i < 3; i++)
        printf("%f\n", b[i]);
  return 0;
}

File: Host/MAGMA/FORTRAN/sgesv.f90
Program LinearEquations
! This solves the matrix equation A*x=b using MAGMA
! Include both the custom magma module (magma.f90) and the ISO_C_BINDING module
! so that the variables that are passed to the magma function are of the right C
! type
  use magma
  use iso_c_binding
  Implicit none

! Declarations, notice single precision
  Real (KIND=C_FLOAT) :: A(3,3), b(3)
  Integer (KIND=C_INT) :: i, j, pivot(3), ok, status

! Define matrix A
  A(1,1)=3.1
  A(1,2)=1.3
  A(1,3)=-5.7
  A(2,1)=1.0
  A(2,2)=-6.9
  A(2,3)=5.8
  A(3,1)=3.4
  A(3,2)=7.2
  A(3,3)=-8.8

! Define vector b, make b a matrix and you can solve multiple
! equations with the same A but different b
  b(1)=-1.3
  b(2)=-0.1
  b(3)=1.8

! Find the solution using the MAGMA routine MAGMA_SGESV
  status = MAGMA_SGESV(3, 1, A, 3, pivot, b, 3, ok)

! Parameters in the order as they appear in the function call
!    order of matrix A, number of right hand sides (b), matrix A,
!    leading dimension of A, array that records pivoting, 
!    result vector b on entry, x on exit, leading dimension of b
!    return value   

! print the vector x
  do i=1, 3
     write(*,*) b(i)
  end do
end
